iothub/device/src/TelemetryMethods.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Shared;
+using Microsoft.Win32;
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    internal static partial class TelemetryMethods
+    {
+        public static string GetSqmMachineId()
+        {
+            try
+            {
+                var key = Registry.LocalMachine.OpenSubKey(@"Software\Microsoft\SQMClient");
+                if (key != null)
```

Comment:
What is the behavior on linux ?

Reply:
I've added a condition to only try getting the reg key value on Windows.  For all other operating systems it will return null.  I've also added a unit test and ran the test/build on Linux.